datafusion/core/tests/sqllogictests/test_files/interval.slt

```diff
@@ -23,7 +23,113 @@ select
   arrow_typeof(interval '5 months'),
   arrow_typeof(interval '5 days 3 nanoseconds')
 ----
-Interval(YearMonth) Interval(MonthDayNano)
+Interval(MonthDayNano) Interval(MonthDayNano)
+
+
+## This is incredibly confusing but document it in tests:
+#
+# years is parsed as a column name
```

Comment:
Is this a bug?

Reply:
DataFusion is consistent with postgres so in that sense it is not a bug 

This behavior is due to how the statement is parsed (sqlparser-rs treats `years` as the alias)

It is incredibly confusing (I was confused for a while)

Now that we have proper `Utf8` -> `Interval` parsing (thanks to @doki23 ) in arrow-rs I think the preferred way to create intervals should be `'5 months'::interval` which does the right thing and doesn't have the wacky syntax